Automattic/jetpack
# RNA: Show compact value on StatCard component

Closes #26040

#### Changes proposed in this Pull Request:

* Add options to `numberFormat` helper function
* Format value in compact form on `square` variant of `StatCard` component

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Open Storybook with `cd projects/js-packages/storybook && npm run storybook:dev`
* On `JS Packages/Components/Stat Card`, verify that the square variant shows the compact value, with the full value as a tooltip
* Verify that the horizontal variant shows the full value directly
* Run `jetpack test js-packages/components js` and confirm all tests passes

![Screenshot_2022-09-06_12-14-09](https://user-images.githubusercontent.com/8486249/188674515-a4cc15c3-088a-4a41-80ac-385f148c9582.png)
